master/internal/user/service.go

```diff
@@ -291,6 +291,76 @@ func (s *Service) patchUser(c echo.Context) (interface{}, error) {
 	}, nil
 }
 
+func (s *Service) patchUsername(c echo.Context) (interface{}, error) {
+	type (
+		request struct {
+			NewUsername *string `json:"username,omitempty"`
+		}
+		response struct {
+			message string
+		}
+	)
+
+	body, err := ioutil.ReadAll(c.Request().Body)
+	if err != nil {
+		return nil, err
+	}
+
+	args := struct {
+		Username string `path:"username"`
+	}{}
+	if err = api.BindArgs(&args, c); err != nil {
+		return nil, err
+	}
+
+	var params request
+	if err = json.Unmarshal(body, &params); err != nil {
+		malformedRequestError := echo.NewHTTPError(http.StatusBadRequest, "bad request")
+		return nil, malformedRequestError
+	}
+
+	forbiddenError := echo.NewHTTPError(http.StatusForbidden)
+	authenticatedUser := c.(*context.DetContext).MustGetUser()
+	if !authenticatedUser.Admin {
+		return nil, forbiddenError
+	}
+
+	user, err := s.db.UserByUsername(args.Username)
+	switch err {
+	case nil:
+	case db.ErrNotFound:
+		if authenticatedUser.Admin {
```

Comment:
This was the logic I was suggesting could be simplified now that we've moved the `if !authenticatedUser.Admin` check up above.

Reply:
ah, yeah I see, sorry was a little fast this morning trying to close it.

Reply:
https://github.com/determined-ai/determined/pull/702